graphql_compiler/compiler/metadata.py

```diff
@@ -166,14 +168,12 @@ def get_filter_infos(self, location):
 
     def record_recurse_info(self, location, recurse_info):
         """Record recursion information about the location."""
-        if location in self._recurse_infos:
-            raise AssertionError(u'Cannot register RecurseInfo twice at same location: '
-                                 u'{}'.format(location))
-        self._recurse_infos[location] = recurse_info
+        record_location = location.at_vertex()
+        self._recurse_infos.setdefault(record_location, []).append(recurse_info)
```

Comment:
Yes, this was my suggestion -- I personally dislike `defaultdict` and prefer not to use it since it makes things that look like reads actually be writes. I think `setdefault()` is obviously a write and not that much more difficult to type out, and I don't like the idea of `foo[bar]` modifying my `foo`.